### DISCUSSION TITLE
Using Liveblocks for collaboration

### DISCUSSION THREAD

**Discussion by itsnitinr** [excalidraw/excalidraw#8305] [Q&A]
Hello, I am trying to add collaboration to the `<Excalidraw />` component from the `@excalidraw/excalidraw` package in a Next.js app. I figured that since I'm using Liveblocks in my app anyway, I would want to use it for collaboration and storage rather than spinning up a separate collaboration server as discussed in the other discussions. Here's a quick boilerplate code of how it looks like:

```tsx
const ExcalidrawCanvas = () => {
  // This is the useStorage hook from Liveblocks
  // nodes gets updated whenever someone add or modifies something in the canvas
  // nodes is a LiveList
  const nodes = useCanvasStorage((storage) => storage.data.nodes);

  const setCanvas = useCanvasMutation(({ storage }, elements) => {
    const nodes = storage.get('data');
    nodes.set('nodes', elements);
  }, []);

  return (
      <Excalidraw
        initialData={{
          elements: nodes || [],
        }}
        onChange={(elements) => {
            setCanvas(elements);
        }}
      />
  );
};
```

My question is if there's a way to programatically update the scene with my nodes from Liveblocks so that I can enable collaboration. 

If this is not possible at the moment, how do I enable collaboration? Will I need to host excalidraw-room, clone excalidraw, change the websocket url in the env, rebuild and publish a new package and use that? Any help is appreciated.

**Comment by Mrazator** [comment]
It sounds like you want to call our `updateScene` API.

https://docs.excalidraw.com/docs/@excalidraw/excalidraw/api/props/excalidraw-api#updatescene
